pkg/metrics/metrics.go

```diff
@@ -28,22 +28,27 @@ import (
 	psnet "github.com/shirou/gopsutil/net"
 )
 
-type Metrics struct {
-	prevBusy float32
-	prevAll  float32
+// A metrics provider gather system and unit information on the host system and
+// return a mapping of the successfully processed metrics.
+type MetricsProvider interface {
+	ReadSystemMetrics(string) api.ResourceMetrics
+	ReadUnitMetrics(string) api.ResourceMetrics
```

Comment:
I use “Read” Instead of “Get” in the method names to indicate that they are doing more than reading in-memory attribute.